lms/djangoapps/discussion/rest_api/views.py

```diff
@@ -68,41 +68,28 @@
 @view_auth_classes()
 class CourseView(DeveloperErrorViewMixin, APIView):
     """
-    **Use Cases**
+    General discussion metadata API.
+    """
 
+    @apidocs.schema(
+        parameters=[
+            apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")
+        ],
+        responses={
+            200: CourseMetadataSerailizer(read_only=True, required=False),
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        }
+    )
+    def get(self, request, course_id):
+        """
         Retrieve general discussion metadata for a course.
 
-    **Example Requests**:
-
-        GET /api/discussion/v1/courses/course-v1:ExampleX+Subject101+2015
-
-    **Response Values**:
-
-        * id: The identifier of the course
-
-        * blackouts: A list of objects representing blackout periods (during
-            which discussions are read-only except for privileged users). Each
-            item in the list includes:
-
-            * start: The ISO 8601 timestamp for the start of the blackout period
-
-            * end: The ISO 8601 timestamp for the end of the blackout period
-
-        * thread_list_url: The URL of the list of all threads in the course.
-
-        * following_thread_list_url: thread_list_url with parameter following=True
+        **Example Requests**:
 
-        * topics_url: The URL of the topic listing for the course.
-
-        * allow_anonymous: A boolean which indicating whether anonymous posts
-            are allowed or not.
-
-        * allow_anonymous_to_peers: A boolean which indicating whether posts
-            anonymous to peers are allowed or not.
-    """
```

Comment:
The response values being documented here would be quite useful for future developers, especially for the response fields added in this PR, as they not exactly course details, but user details wrt the course.  Therefore these should not be removed in my view.

Reply:
I have moved this documentation to the seraialiser, and as such it will now appear at http://localhost:18000/api-docs/ as a unified place for documentation. Documenting it this way makes it compatible with the DRF-yasg which generated API documentation that's compatible with Open API standards.